utils/python/CIME/compare_test_results.py

```diff
@@ -0,0 +1,86 @@
+import CIME.compare_namelists, CIME.simple_compare
+from CIME.utils import expect
+from CIME.XML.machines import Machines
+from CIME.test_status import *
+from CIME.hist_utils import compare_baseline
+from CIME.case import Case
+
+import os, glob
+
+_MACHINE = Machines()
+
+###############################################################################
+def compare_history(testcase_dir_for_test, baseline_name):
+###############################################################################
+    with Case(testcase_dir_for_test) as case:
+        baseline_full_dir = os.path.join(case.get_value("BASELINE_ROOT"), case.get_value("COMPILER"), baseline_name, case.get_value("CASEBASEID"))
```

Comment:
I can't answer that myself. It would affect many people in CSEG, so I imagine needs some broader buy-in. Personally, I can't think of any major objections, but I also don't see the advantage, since our test names have the compiler in them. So it seems to work just fine to have a baseline directory that contains, say, SMS.f10_f10.ICLM45.yellowstone_intel, SMS.f10_f10.ICLM45.yellowstone_pgi and SMS.f10_f10.ICLM45.yellowstone_gnu.

I'll defer to @mvertens , @jedwards4b and other CSEG members on this question.

I do see that there are other places in the scripts that assume the ACME organization in this respect: at least bless_test_results.py and maybe test_scheduler.py and scripts_regression_tests.py (I just did a simple grep for grep -i 'path.join.*compiler').